storage/watchable_store_test.go

```diff
@@ -222,3 +240,130 @@ func TestUnsafeAddWatching(t *testing.T) {
 		}
 	}
 }
+
+func TestNewMapWatchingToEventMap(t *testing.T) {
+	var (
+		testN         = 5
+		testKey       = []byte("Foo")
+		testValue     = []byte("Bar")
+		testKeys      = make([][]byte, testN)
+		testValues    = make([][]byte, testN)
+		testWatchings = make([]*watching, testN)
+		testEvents    = make([]storagepb.Event, testN)
+	)
+	for i := 0; i < testN; i++ {
+		k, v := testKey, testValue
+		if i > 0 {
+			k = []byte(fmt.Sprintf("%s_%d", testKey, i))
+			v = []byte(fmt.Sprintf("%s_%d", testValue, i))
+		}
+		testKeys[i] = k
+		testValues[i] = v
+		testWatchings[i] = &watching{
+			key:    k,
+			prefix: true,
+		}
+		testEvents[i] = storagepb.Event{
+			Type: storagepb.PUT,
+			Kv: &storagepb.KeyValue{
+				Key:   k,
+				Value: v,
+			},
+		}
+	}
+
```

Comment:
@xiang90 Just added different test cases. Please take a look again and let me know if you have any other feedback.